acmw-db-app/components/FormComponents/CompanySearchInput.jsx

```diff
@@ -1,12 +1,54 @@
 import SelectInput from './SelectInput'
+import {useState} from 'react'
+import styles from '../../styles/components/FormComponents.module.css'
+import MultiSelectInput from './MultiSelectInput'
 
-const CompanySearchInput = () => {
+// I'm so bad at understanding promises/fetch() so please point out if I'm doing something wrong
 
-    // Eventually this will connect to the database of companies and provide a full list
-    const sampleCompanies = ["Nimbis Services", "Target", "Capital One", "Facebook"]
+const getCompanyInfo = async (input) => {
+    const url = '/api/company/byString?input=' + input
+    const response = await fetch(url, {
+        method: 'GET', 
+        mode: 'cors', 
+        cache: 'no-cache', 
+        credentials: 'same-origin', 
+        headers: {
+        'Content-Type': 'application/json'
+        },
+        redirect: 'follow', 
+        referrerPolicy: 'no-referrer', 
+    })
+    return response.json() 
+}
+
+const CompanySearchInput = (props) => {
+
+    const [companyList, setCompanyList] = useState([])
+    const [isText, setIsText] = useState(false) // Has the user entered text
+
+    const filterFunction = (event) => {
+        if (event.target.value != "") {
```

Comment:
Probs safer to use `if (event.target.value)` to check if not empty/null/undefined